garden_ai/client.py

```diff
@@ -458,15 +458,37 @@ def get_published_garden(self, doi: str) -> Garden:
         return garden
 
     def _generate_presigned_urls_for_garden(self, garden: Garden):
+        all_model_names = [
+            model_name
+            for pipeline in garden.pipelines
+            for model_name in pipeline.model_full_names
+        ]  # flatten all model names
+        all_presigned_urls = self.backend_client.get_model_download_urls(
+            all_model_names
+        )
         for pipeline in garden.pipelines:
-            pipeline_url_json = self.generate_presigned_urls_for_pipeline(pipeline)
-            pipeline._env_vars = {GardenConstants.URL_ENV_VAR_NAME: pipeline_url_json}
+            model_name_to_url = {}
+            for model_name in pipeline.model_full_names:
+                model_name_to_url[model_name] = next(
+                    presigned_url.url
+                    for presigned_url in all_presigned_urls
+                    if presigned_url.model_name == model_name
+                )
```

Comment:
Not a blocker, but I feel like there's a more straightforward way to express this.